### ISSUE TITLE
Add intersection types

### ISSUE THREAD

**Issue by zerkalica** [gcanti/tcomb#111]
For building complex validation rules.

```js
    var MyStr = t.intersect([Str, NotEmptyStr, MinStr(5)])
    MyStr.is(123) // Str
    MyStr.is('') // NotEmptyStr
    MyStr.is('123') // MinStr: 5
    MyStr.is('12345') // ok
```

**Comment by zerkalica** [comment]
I rewrote you tcomb-validator https://github.com/zerkalica/tcomb-validator-plus

Now it supports rules like this:

```js
const Password = t.intersect([
  t.alias(t.StrNotEmpty, 'password is empty'),
  t.alias(t.StrMin(3), 'password less than #{min}'),
  t.alias(t.StrMax(5), 'password greate than #{max}')
], 'Password')
```

And can be extended to support localizations, like babelfish or node-polyglot:

```js
class MyValidator extends Validator {
  constructor(t) {
    super()
    this._t = t
  }

  _formatMessage(message, params) {
    return this._t(message, params)
  }
}
```

**Comment by gcanti** [comment]
Hi @zerkalica
Sorry for being so late, the `intersection` combinator is definitely interesting, I'll be working on this for v2.2

**Comment by gcanti** [comment]
Hi,
I've just released v2.2.0 with a new intersection combinator

https://github.com/gcanti/tcomb/releases/tag/v2.2.0
